x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -34,6 +35,11 @@ impl From<Pointer> for *mut u8 {
 /// and the following 4 bytes represent a pointer to WASM memeory where the bytes are stored.
 pub type HostPtr = i64;
```

Comment:
I thought about making it `#[deprecated]`, but will probably just have to remove it in this PR because the go part is broken since it still rely a bit on the `SmartPtr`

Reply:
Since you posted, I think we solved the Go side. You can remove this in a follow up